lib/tic_tac_toe.ex

```diff
@@ -0,0 +1,95 @@
+defmodule TicTacToe do
+  
+  def new_game do
+    %Game{}
+  end
+
+  def start do
+    %Message{}.welcome |> Console.print
+    Board.create |> Console.print_board
+    new_game() |> loop
+  end
+
+  def loop(game) do
+    game = 
+      case game.current_player == %Player{} do
+        true -> human_turn(game)
+        false -> computer_turn(game)
+      end
+
+    if Game.game_over?(game) do
+      %Message{}.game_over |> Console.print
+      Game.get_winner(game).winner |> win_message
+      Console.play_again |> restart_game
+    else 
+      loop(game) 
+    end
+  end
```

Comment:
Just a thought, if you are interested in making this testable, you could tease out the `game =` expression into one function, and then the `if Game.game_over?` section could be a second function, which _accepts a function_.

This notation might not be perfectly valid syntax, but just as an example:

```elixir
def perform_game_turn(game, continue_game) do
  if Game.game_over?(game) do
    ...
  else
    continue_game.(game)
  end
end

def loop(game) do
  game = ...
  perform_game_turn(game, &loop/1)
end
```